File: app/src/main/java/dev/pitlor/smssync/db/OwnCloud.java
package dev.pitlor.smssync.db;

import com.thegrizzlylabs.sardineandroid.Sardine;
import com.thegrizzlylabs.sardineandroid.impl.OkHttpSardine;

public class OwnCloud {
    private static Sardine instance;

    public static Sardine getInstance() {
        if (instance == null) {
            instance = new OkHttpSardine();
            instance.setCredentials("admin", "admin");
        }

        return instance;
    }
}

File: app/src/main/java/dev/pitlor/smssync/datasources/OwnCloud.kt
package dev.pitlor.smssync.datasources

import com.thegrizzlylabs.sardineandroid.Sardine
import com.thegrizzlylabs.sardineandroid.impl.OkHttpSardine

object OwnCloud {
    var instance: Sardine? = null
        get() {
            if (field == null) {
                field = OkHttpSardine()
                (field as OkHttpSardine).setCredentials("admin", "admin")
            }
            return field
        }
        private set
}